block-storage/src/main/scala/coop/rchain/blockstorage/FileLMDBIndexBlockStore.scala

```diff
@@ -55,24 +70,54 @@ class FileLMDBIndexBlockStore[F[_]: Monad: Sync] private (
   private[this] def withReadTxn[R](f: Txn[ByteBuffer] => R): F[R] =
     withTxn(env.txnRead())(f)
 
-  private def readBlockMessage(offset: Long): F[BlockMessage] =
+  private def readBlockMessage(indexEntry: IndexEntry): F[BlockMessage] = {
+    def readBlockMessageFromFile(storageFile: RandomAccessFile): F[BlockMessage] =
+      for {
+        _                      <- Sync[F].delay { storageFile.seek(indexEntry.offset) }
+        blockMessageSize       <- Sync[F].delay { storageFile.readInt() }
+        blockMessagesByteArray = Array.ofDim[Byte](blockMessageSize)
+        _                      <- Sync[F].delay { storageFile.readFully(blockMessagesByteArray) }
+        blockMessage           = BlockMessage.parseFrom(blockMessagesByteArray)
```

Comment:
What happens if this parseFrom fails? Does the exception bubble up and magically become a None result on the BlockStore.get call?